fwdpy11/genetic_map_unit.py

```diff
@@ -66,9 +66,12 @@ class PoissonInterval(fwdpy11._fwdpy11._ll_PoissonInterval):
     beg: float
     end: float
     mean: float
+    discrete: bool = False
 
     def __attrs_post_init__(self):
-        super(PoissonInterval, self).__init__(self.beg, self.end, self.mean)
+        super(PoissonInterval, self).__init__(
```

Comment:
All of these post-init functions need to:

1. Check that beg/end/position are indeed integer if discrete is True.
1. Check that end - beg > 1 if discrete is True.  If not, we should error out, as the user is modelling a "point" and not an "interval".

Reply:
We also need tests that the above checks raise errors.

Reply:
Circle CI is failing b/c of old pybind11.  We need to change to require >= 2.6.0.  Should do this in a separate PR and rebase this one.

Reply:
May want to handle molpopgen/fwdpp#325 and update the submodule here?

Reply:
Type hints need changing here, from float to "numeric".